oracle/src/main.sw

```diff
@@ -0,0 +1,58 @@
+contract;
+
+dep interface;
+dep data_structures;
+dep errors;
+dep events;
+
+use std::{
+    chain::auth::msg_sender,
+    constants::BASE_ASSET_ID,
+    identity::Identity,
+    logging::log,
+    result::Result,
+    revert::require,
+};
+
+use interface::Oracle;
+use data_structures::State;
+use errors::{AccessError, InitializationError};
+use events::PriceUpdateEvent;
+
+storage {
+    /// The Identity that can control the oracle (node)
+    owner: Identity = Identity::ContractId(BASE_ASSET_ID),
+    /// Current price of tracked asset
+    price: u64 = 0,
+    /// The initialization state of the contract.
+    state: State = State::NotInitialized,
+}
+
+impl Oracle for Contract {
+    #[storage(read, write)] fn constructor(owner: Identity) {
```

Comment:
Maybe instead of a constructor here, consider a configuration-time constant for the owner? This is a brand new feature and not documented yet but you can find some details in the PR :

https://github.com/FuelLabs/sway/pull/2549